src/stream/mod.rs

```diff
@@ -250,7 +250,40 @@ pub trait Stream {
     fn boxed(self) -> BoxStream<Self::Item, Self::Error>
         where Self: Sized + Send + 'static,
     {
-        ::std::boxed::Box::new(self)
+        #[cfg(rust_at_least_1_18)]
+        fn boxed<S>(stream: S) -> BoxStream<S::Item, S::Error>
```

Comment:
This seems to be a fully copy/paste version of `boxed()` for `Future` except the result type itself. I'd rather decompose them both to a some type-parametrized function.